examples/esplora_backend_with_tor.rs

```diff
@@ -0,0 +1,178 @@
+use std::fs::OpenOptions;
+use std::io::Read;
+use std::str::FromStr;
+use std::time::Duration;
+use std::{env, thread};
+
+use bitcoin::{util::bip32, Network};
+use esplora_client::Builder;
+use libtor::{LogDestination, LogLevel, Tor, TorFlag};
+
+use bdk::{
+    blockchain::esplora::EsploraBlockchain, database::MemoryDatabase, template::Bip84,
+    KeychainKind, SyncOptions, Wallet,
+};
+use bdk::wallet::AddressIndex;
+
+pub mod utils;
+
+use crate::utils::tx::build_signed_tx;
+
+/// This will create a wallet from an xpriv and sync it by connecting to an Esplora server
+/// over Tor network, using blocking calls with `ureq`.
+///
+/// This can be run with `cargo run --features="use-esplora-blocking libtor" --example esplora_backend_with_tor`
+/// in the root folder.
+///
+/// Note, that libtor requires a C compiler and C build tools.
+///
+/// Here is how to install them on Ubuntu:
+/// ```
+/// sudo apt install autoconf automake clang file libtool openssl pkg-config
+/// ```
+/// And here is how to install them on Mac OS X:
+/// ```
+/// xcode-select --install
+/// brew install autoconf
+/// brew install automake
+/// brew install libtool
+/// brew install openssl
+/// brew install pkg-config
+/// export LDFLAGS="-L/opt/homebrew/opt/openssl/lib"
+/// export CPPFLAGS="-I/opt/homebrew/opt/openssl/include"
+/// ```
+fn main() {
+    let network = Network::Testnet;
+
+    let xpriv_str =
+        "tprv8ZgxMBicQKsPcx5nBGsR63Pe8KnRUqmbJNENAfGftF3yuXoMMoVJJcYeUw5eVkm9WBPjWYt6HMWYJNesB5HaNVBaFc1M6dRjWSYnmewUMYy";
+
+    let esplora_url =
+        "http://explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion/testnet/api";
+
+    let xpriv = bip32::ExtendedPrivKey::from_str(xpriv_str).unwrap();
+
+    let socks_addr = start_tor();
+
+    let client = Builder::new(esplora_url)
+        .proxy(&format!("socks5://{}", socks_addr))
+        .timeout(120) // seconds
+        .build_blocking()
+        .unwrap();
+
+    println!(
+        "Connecting to {} via SOCKS5 proxy {}",
+        &esplora_url, &socks_addr
+    );
+
+    let blockchain = EsploraBlockchain::from_client(client, 20);
```

Comment:
This is a valid way of building the client, but not the intended API for bdk for such cases. 
Users using bdk, should not have to use `elsplora_client` in their dependency. Handling `esplora-client` is already done internally in bdk. 

`EsploraBlockhain` in bdk implements `ConfigurableBlockchain` which has an API `from_config`. https://github.com/bitcoindevkit/bdk/blob/eac739d3954937e49705a7336b4a017652802996/src/blockchain/esplora/blocking.rs#L215-L218

So you can create an esplora config using this structure https://github.com/bitcoindevkit/bdk/blob/eac739d3954937e49705a7336b4a017652802996/src/blockchain/esplora/mod.rs#L36-L62

Fields are public, so you can just set individual values.. 

And then just call `EsploraBlockhain::from_config(&config)`.

Reply:
👍